src/lib/Hydra/Controller/Root.pm

```diff
@@ -15,6 +15,9 @@ use JSON;
 use List::Util qw[min max];
 use List::MoreUtils qw{any};
 use Net::Prometheus;
+use Types::Standard qw/StrMatch/;
+
+use constant NARINFO_REGEX => qr{^([a-z0-9]+)\.narinfo$};
```

Comment:
Would be nice if we could make this match the length of the hash as well, but adding `{32}` made the server hang or something (in my brief testing).

Reply:
Hmm... how did you try for it? I used: 

```
use constant NARINFO_REGEX => qr{^([a-z0-9]{32})\.narinfo$};
```

and it seemed good from here. If we wanted to be Very Precise we could use the exact character set of a store path: 0123456789abcdfghijklmnpqrsvwxyz but we probably don't need to be.

Reply:
Heh, whoops. I think I placed it OUTSIDE the capture group.